transmogrifier/sources/xml/marc.py

```diff
@@ -895,6 +793,167 @@ def _get_links_holdings_electronic_items(
                     text=holding_collection,
                 )
 
+    @classmethod
+    def get_literary_form(cls, source_record: Tag) -> str | None:
+        """Retrieve literary form for book materials.
+
+        Book materials configurations are used when Leader/06 (Type of record)
+        contains code a (Language material) or t (Manuscript language material)
+        and Leader/07 (Bibliographic level) contains code
+        a (Monographic component part), c (Collection), d (Subunit),
+        or m (Monograph).
+        """
+        leader_field = cls._get_leader_field(source_record)
+        control_field = cls._get_control_field(source_record)
+        if leader_field[6] in "at" and leader_field[7] in "acdm":
+            if control_field[33] in "0se":
+                return "Nonfiction"
+            return "Fiction"
+        return None
+
+    @classmethod
+    def get_locations(cls, source_record: Tag) -> list[timdex.Location] | None:
+        locations = []
+        location_marc_fields = [
+            {
+                "tag": "751",
+                "subfields": "a",
+                "kind": "Geographic Name",
+            },
+            {
+                "tag": "752",
+                "subfields": "abcdefgh",
+                "kind": "Hierarchical Place Name",
+            },
+        ]
+
+        # get locations (place of publication) from control field 008/15-17
+        if (
+            fixed_location_code := cls._get_control_field(source_record)[15:18].strip()
+        ) and (
+            location_name := Marc.loc_crosswalk_code_to_name(
```

Comment:
Same here, think could be `cls.loc_crosswalk_code_to_name()`.

Reply:
Updated!